TST: Calling nonzero on 0d arrays is deprecated

This failure started appearing in the `numpy-dev` job. Example log: https://travis-ci.org/astropy/astropy/jobs/563545076

```
______________________________ test_bounding_box _______________________________
    def test_bounding_box():
        g = Gaussian2D() + Gaussian2D(2, .5, .1, 2, 3, 0)
        g.bounding_box = ((0, 1), (0, .5))
        y, x = np.mgrid[0:10, 0:10]
        y = y / 3.
        x = x / 3.
        val = g(x, y, with_bounding_box=True)
        compare = np.array([
            [2.93738984, 2.93792011, np.nan, np.nan, np.nan,
             np.nan, np.nan, np.nan, np.nan, np.nan],
            [2.87857153, 2.88188761, np.nan, np.nan, np.nan,
             np.nan, np.nan, np.nan, np.nan, np.nan],
            [2.70492922, 2.71529265, np.nan, np.nan, np.nan,
             np.nan, np.nan, np.nan, np.nan, np.nan],
            [2.45969972, 2.47912103, np.nan, np.nan, np.nan,
             np.nan, np.nan, np.nan, np.nan, np.nan],
            [np.nan, np.nan, np.nan, np.nan, np.nan,
             np.nan, np.nan, np.nan, np.nan, np.nan],
            [np.nan, np.nan, np.nan, np.nan, np.nan,
             np.nan, np.nan, np.nan, np.nan, np.nan],
            [np.nan, np.nan, np.nan, np.nan, np.nan,
             np.nan, np.nan, np.nan, np.nan, np.nan],
            [np.nan, np.nan, np.nan, np.nan, np.nan,
             np.nan, np.nan, np.nan, np.nan, np.nan],
            [np.nan, np.nan, np.nan, np.nan, np.nan,
             np.nan, np.nan, np.nan, np.nan, np.nan],
            [np.nan, np.nan, np.nan, np.nan, np.nan,
             np.nan, np.nan, np.nan, np.nan, np.nan]])
        mask = ~np.isnan(val)
        assert_allclose(val[mask], compare[mask])
        val2 = g(x+2, y+2, with_bounding_box=True)
        assert(np.isnan(val2).sum() == 100)
>       val3 = g(.1, .1, with_bounding_box=True)
astropy/modeling/tests/test_compound.py:555: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astropy/modeling/core.py:2441: in __call__
    self, input_shape, args, bbox)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <CompoundModel(amplitude_0=1., x_mean_0=0., y_mean_0=0., x_stddev_0=1., y_stddev_0=1., theta_0=0., amplitude_1=2., x_mean_1=0.5, y_mean_1=0.1, x_stddev_1=2., y_stddev_1=3., theta_1=0.)>
input_shape = (), inputs = (0.1, 0.1), bbox = ((0, 0.5), (0, 1))
    def prepare_bounding_box_inputs(self, input_shape, inputs, bbox):
    
        allout = False
        if self.n_inputs > 1:
            # bounding_box is in python order -
            # convert it to the order of the inputs
            bbox = bbox[::-1]
        if self.n_inputs == 1:
            bbox = [bbox]
        # indices where input is outside the bbox
        # have a value of 1 in ``nan_ind``
        nan_ind = np.zeros(input_shape, dtype=bool)
        for ind, inp in enumerate(inputs):
            inp = np.asanyarray(inp)
            outside = np.logical_or(inp < bbox[ind][0], inp > bbox[ind][1])
            if inp.shape:
                nan_ind[outside] = True
            else:
                nan_ind |= outside
                if nan_ind:
                    allout = True
        # get an array with indices of valid inputs
>       valid_ind = np.logical_not(nan_ind).nonzero()
E       DeprecationWarning: Calling nonzero on 0d arrays is deprecated, as it behaves surprisingly. Use `atleast_1d(cond).nonzero()` if the old behavior was intended. If the context of this warning is of the form `arr[nonzero(cond)]`, just use `arr[cond]`.
astropy/modeling/core.py:3955: DeprecationWarning
____________________________ test_with_bounding_box ____________________________
    def test_with_bounding_box():
        """
        Test the option to evaluate a model respecting
        its bunding_box.
        """
        p = models.Polynomial2D(2) & models.Polynomial2D(2)
        m = models.Mapping((0, 1, 0, 1)) | p
        with NumpyRNGContext(1234567):
            m.parameters = np.random.rand(12)
    
        m.bounding_box = ((3, 9), (1, 8))
        x, y = np.mgrid[:10, :10]
        a, b = m(x, y)
        aw, bw = m(x, y, with_bounding_box=True)
        ind = (~np.isnan(aw)).nonzero()
        assert_allclose(a[ind], aw[ind])
        assert_allclose(b[ind], bw[ind])
    
        aw, bw = m(x, y, with_bounding_box=True, fill_value=1000)
        ind = (aw != 1000).nonzero()
        assert_allclose(a[ind], aw[ind])
        assert_allclose(b[ind], bw[ind])
    
        # test the order of bbox is not reversed for 1D models
        p = models.Polynomial1D(1, c0=12, c1=2.3)
        p.bounding_box = (0, 5)
        assert(p(1) == p(1, with_bounding_box=True))
    
        t3 = models.Shift(10) & models.Scale(2) & models.Shift(-1)
        t3.bounding_box = ((4.3, 6.9), (6, 15), (-1, 10))
        assert_allclose(t3([1, 1], [7, 7], [3, 5],with_bounding_box=True),
                        [[np.nan, 11], [np.nan, 14], [np.nan, 4]])
    
        trans3 = models.Shift(10) & models.Scale(2) & models.Shift(-1)
        trans3.bounding_box = ((4.3, 6.9), (6, 15), (-1, 10))
>       assert_allclose(trans3(1, 7, 5, with_bounding_box=True), [11, 14, 4])
astropy/modeling/tests/test_models.py:683: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astropy/modeling/core.py:2441: in __call__
    self, input_shape, args, bbox)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <CompoundModel(offset_0=10., factor_1=2., offset_2=-1.)>
input_shape = (), inputs = (1, 7, 5), bbox = ((-1, 10), (6, 15), (4.3, 6.9))
    def prepare_bounding_box_inputs(self, input_shape, inputs, bbox):
    
        allout = False
        if self.n_inputs > 1:
            # bounding_box is in python order -
            # convert it to the order of the inputs
            bbox = bbox[::-1]
        if self.n_inputs == 1:
            bbox = [bbox]
        # indices where input is outside the bbox
        # have a value of 1 in ``nan_ind``
        nan_ind = np.zeros(input_shape, dtype=bool)
        for ind, inp in enumerate(inputs):
            inp = np.asanyarray(inp)
            outside = np.logical_or(inp < bbox[ind][0], inp > bbox[ind][1])
            if inp.shape:
                nan_ind[outside] = True
            else:
                nan_ind |= outside
                if nan_ind:
                    allout = True
        # get an array with indices of valid inputs
>       valid_ind = np.logical_not(nan_ind).nonzero()
E       DeprecationWarning: Calling nonzero on 0d arrays is deprecated, as it behaves surprisingly. Use `atleast_1d(cond).nonzero()` if the old behavior was intended. If the context of this warning is of the form `arr[nonzero(cond)]`, just use `arr[cond]`.
astropy/modeling/core.py:3955: DeprecationWarning
```

cc @perrygreenfield , @nden , and @mhvk

--- a/astropy/modeling/core.py
+++ b/astropy/modeling/core.py
@@ -3952,7 +3952,7 @@ def prepare_bounding_box_inputs(self, input_shape, inputs, bbox):
             if nan_ind:
                 allout = True
     # get an array with indices of valid inputs
-    valid_ind = np.logical_not(nan_ind).nonzero()
+    valid_ind = np.atleast_1d(np.logical_not(nan_ind)).nonzero()
     if len(valid_ind[0]) == 0:
         allout = True
     # inputs holds only inputs within the bbox

